src/irust/cursor.rs

```diff
@@ -3,121 +3,136 @@ use crate::utils::StringTools;
 mod bounds;
 use bounds::Bounds;
 
+use crossterm::TerminalCursor;
+
 #[derive(PartialEq)]
 pub enum Move {
     Free,
     Modify,
 }
 
 #[derive(Clone)]
-pub struct Cursor {
+pub struct CursorPosition {
     pub screen_pos: (usize, usize),
     pub buffer_pos: usize,
     pub bounds: Bounds,
     pub lock_pos: (usize, usize),
+}
+
+pub struct Cursor {
+    pub pos: CursorPosition,
+    internal_cursor: TerminalCursor,
     origin: (usize, usize, usize),
-    copy: Option<Box<Cursor>>,
+    copy: Option<Box<CursorPosition>>,
 }
+
 impl Cursor {
-    pub fn new(x: usize, y: usize, width: usize) -> Self {
+    pub fn new(internal_cursor: TerminalCursor, x: usize, y: usize, width: usize) -> Self {
```

Comment:
@sigmaSd The reason I have it like that is because of [this line here](https://github.com/smolck/IRust/blob/4986e7718d320616116770e326078ad3fc597386/src/irust.rs#L52), where we get the cursor from the newly created `Crossterm`. Is that an unnecessary extra step?

Reply:
Yeah Its unecessery, crossterm is actually a package of independent peaces, the Crossterm struct is just a nice wrapper, so yeah you can remove it